backend/tilavarauspalvelu/api/graphql/types/recurring_reservation/serializers/reschedule_serializer.py

```diff
@@ -143,6 +145,20 @@ def update(
 
         has_access_codes = instance.reservations.requires_active_access_code().exists()
 
+        # If any reservation had access codes before reschedule, or has access codes after reschedule,
+        # reschedule the series to make sure its up to date.
+        if had_access_codes or has_access_codes:
+            # Allow mutation to succeed if Pindora request fails.
```

Comment:
The mutation should fail if the Pindora reschedule fails (except that `NotFound` is OK -- in which case the **create** call need not succeed), since we don't want to leave reservees with an access code still functioning at the old times (same as time adjust of individual reservation).

Reply:
Current adjust endpoints call `PindoraService.sync_access_code` and will fail if `create_access_code` fails. Should we call `PindoraService.sync_access_code` here so that it works similarly?

```python
if had_access_codes or has_access_codes:
    with external_service_errors_as_validation_errors(code=error_codes.PINDORA_ERROR):
        PindoraService.sync_access_code(obj=instance)
```

Reply:
Changed to sync

Reply:
Agreed; the corner cases aren't really worth making it more complicated.